src/wallet/feebumper.cpp

```diff
@@ -0,0 +1,275 @@
+            if (nOutput != -1) {
+                vErrors.push_back("Transaction has multiple change outputs");
+                currentResult = BumpFeeResult::WALLET_ERROR;
+                return;
+            }
+            nOutput = i;
+        }
+    }
+    if (nOutput == -1) {
+        vErrors.push_back("Transaction does not have a change output");
+        currentResult = BumpFeeResult::WALLET_ERROR;
+        return;
+    }
+
+    // Calculate the expected size of the new transaction.
+    int64_t txSize = GetVirtualTransactionSize(*(wtx.tx));
+    const int64_t maxNewTxSize = CalculateMaximumSignedTxSize(pWallet, *wtx.tx);
+    if (maxNewTxSize < 0) {
+        vErrors.push_back("Transaction contains inputs that cannot be signed");
+        currentResult = BumpFeeResult::INVALID_ADDRESS_OR_KEY;
+        return;
+    }
+
+    // calculate the old fee and fee-rate
+    nOldFee = wtx.GetDebit(ISMINE_SPENDABLE) - wtx.tx->GetValueOut();
+    CFeeRate nOldFeeRate(nOldFee, txSize);
+    CFeeRate nNewFeeRate;
+    // The wallet uses a conservative WALLET_INCREMENTAL_RELAY_FEE value to
+    // future proof against changes to network wide policy for incremental relay
+    // fee that our node may not be aware of.
+    CFeeRate walletIncrementalRelayFee = CFeeRate(WALLET_INCREMENTAL_RELAY_FEE);
+    if (::incrementalRelayFee > walletIncrementalRelayFee) {
+        walletIncrementalRelayFee = ::incrementalRelayFee;
+    }
+
+    if (totalFee > 0) {
+        CAmount minTotalFee = nOldFeeRate.GetFee(maxNewTxSize) + ::incrementalRelayFee.GetFee(maxNewTxSize);
+        if (totalFee < minTotalFee) {
+            vErrors.push_back(strprintf("Insufficient totalFee, must be at least %s (oldFee %s + incrementalFee %s)",
+                                                                FormatMoney(minTotalFee), FormatMoney(nOldFeeRate.GetFee(maxNewTxSize)), FormatMoney(::incrementalRelayFee.GetFee(maxNewTxSize))));
+            currentResult = BumpFeeResult::INVALID_PARAMETER;
+            return;
+        }
+        CAmount requiredFee = pWallet->GetRequiredFee(maxNewTxSize);
+        if (totalFee < requiredFee) {
+            vErrors.push_back(strprintf("Insufficient totalFee (cannot be less than required fee %s)",
+                                                                FormatMoney(requiredFee)));
+            currentResult = BumpFeeResult::INVALID_PARAMETER;
+            return;
+        }
+        nNewFee = totalFee;
+        nNewFeeRate = CFeeRate(totalFee, maxNewTxSize);
+    } else {
+        // if user specified a confirm target then don't consider any global payTxFee
+        if (specifiedConfirmTarget) {
+            nNewFee = pWallet->GetMinimumFee(maxNewTxSize, newConfirmTarget, mempool, CAmount(0));
+        }
+        // otherwise use the regular wallet logic to select payTxFee or default confirm target
+        else {
+            nNewFee = pWallet->GetMinimumFee(maxNewTxSize, newConfirmTarget, mempool);
```

Comment:
Yes. This seems to be again a rebase issue. Though, changing it to `pWallet` would allow to be more flexible when multi wallet takes off... but changing it back for now.